app/src/test/webapp/app/error/error.ctrl.test.js

```diff
@@ -178,4 +193,72 @@ describe("ErrorCtrl", function() {
         });
     });
   });
+
+  describe("success after calling resubmit", function() {
+    it("should display a success message when the message is resubmitted", function() {
+      spyOn(messageCenterService, 'add');
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = resubmittableMessage;
+
+      $scope.resubmitMessage();
+      $scope.$apply();
+
+      expect(messageCenterService.add).toHaveBeenCalledWith('success', 'Resubmit successful!', { "timeout": 3000 });
+    });
+  });
+
+  describe("failure after calling resubmit", function() {
+    it("should display a failure message when a message isn't resubmittable", function(){
+
+      var messageWithoutResubmitHeader = {
+        id: 1
+      };
+
+      spyOn(messageCenterService, 'add');
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = messageWithoutResubmitHeader;
+
+      $scope.resubmitMessage();
+
+      expect(messageCenterService.add).toHaveBeenCalledWith('warning', 'Message can not be resubmitted', {
+        timeout: 5000
+      });
+
+    });
+
+    beforeEach(function failedCallToResubmitBackend() {
+      msgSvc.resubmitMessage = function() {
+        return $q.when({
+          data: {
+            "status": "Failure",
+            "errorMessage": "Just failed"
+          }
+        });
+
+      };
+    });
+
+    it("should display failure message when there's a server problem", function() {
+      spyOn(messageCenterService, 'add');
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = resubmittableMessage;
+
+      $scope.resubmitMessage();
+      $scope.$apply();
+
+      expect(messageCenterService.add).toHaveBeenCalledWith('danger', 'Just failed', { "status": messageCenterService.status.permanent });
+    });
+  });
```

Comment:
Testing the message center service is nice, but I really care that the actual resubmitMessage() API was called on the EsbMessageService. Looking at the tests, it just tells me that we show a message. It doesn't tell me we actually implement the resubmit functionality.